packages/vm/src/runTx.ts

```diff
@@ -505,7 +513,9 @@ async function _runTx(this: VM, opts: RunTxOpts): Promise<RunTxResult> {
   state.clearOriginalStorageCache()
 
   // Generate the tx receipt
-  const cumulativeGasUsed = (opts.blockGasUsed ?? block.header.gasUsed).add(results.gasUsed)
+  const gasUsed = opts.blockGasUsed ?? bnToBigInt(block.header.gasUsed)
```

Comment:
`??` should not be used as per #1729

Reply:
I guess we might find a better solution here, see comment from @ryanio on #1729